test/e2e/test/elasticsearch/checks_es.go

```diff
@@ -185,29 +171,173 @@ func (e *esClusterChecks) CheckESNodesTopology() test.Step {
+	}
+	return nil
+}
+
+// compareCgroupCPULimit compares the CPU limit specified in a nodeSet with the limit set in the CPU control group at the OS level
+func compareCgroupCPULimit(topologyElement esv1.NodeSet, cgroupCPU struct {
+	CFSPeriodMicros int `json:"cfs_period_micros"`
+	CFSQuotaMicros  int `json:"cfs_quota_micros"`
+}) error {
+	var expectedCPULimit *resource.Quantity
+	for _, c := range topologyElement.PodTemplate.Spec.Containers {
+		if c.Name == esv1.ElasticsearchContainerName {
+			expectedCPULimit = c.Resources.Limits.Cpu()
+		}
 	}
+	if expectedCPULimit == nil || expectedCPULimit.IsZero() {
+		// no expected cpu, consider it's ok
+		return nil
+	}
+
+	actualCgroupCPULimit := float64(cgroupCPU.CFSQuotaMicros) / float64(cgroupCPU.CFSPeriodMicros)
+	if expectedCPULimit.AsApproximateFloat64() != actualCgroupCPULimit {
+		return fmt.Errorf("expected Cgroup CPU limit [%f], got [%f]", expectedCPULimit.AsApproximateFloat64(), actualCgroupCPULimit)
+	}
+	return nil
+}
+
+// compareSpecResources compares the resources specified in the Elasticsearch resource with the resources
+// specified in the pods
+func compareSpecResources(topologyElement esv1.NodeSet, pods []corev1.Pod) error {
+	var expected *corev1.ResourceRequirements
+	for _, c := range topologyElement.PodTemplate.Spec.Containers {
+		container := c
+		if c.Name == esv1.ElasticsearchContainerName {
+			expected = &container.Resources
+		}
+	}
+	if expected == nil {
+		expected = &nodespec.DefaultResources
+	}
+
+	for _, pod := range pods {
+		for _, c := range pod.Spec.Containers {
+			actual := c.Resources
+			if c.Name == esv1.ElasticsearchContainerName { //nolint:nestif
+				if expected.Requests != nil {
+					if err := compareQuantity("CPU request", expected.Requests.Cpu(), actual.Requests.Cpu()); err != nil {
+						return err
+					}
+					if err := compareQuantity("memory request", expected.Requests.Memory(), actual.Requests.Memory()); err != nil {
+						return err
+					}
+				}
+				if expected.Limits != nil {
+					if err := compareQuantity("CPU limit", expected.Limits.Cpu(), actual.Limits.Cpu()); err != nil {
+						return err
+					}
+					if err := compareQuantity("memory limit", expected.Limits.Memory(), actual.Limits.Memory()); err != nil {
+						return err
+					}
+				}
+			}
+		}
+	}
+	return nil
+}
 
-	expectedBytes := memoryLimit.Value()
+func compareQuantity(msg string, expected, actual *resource.Quantity) error {
+	if !expected.IsZero() && !equality.Semantic.DeepEqual(expected, actual) {
+		return fmt.Errorf("expected %s [%d], got [%d]", msg, expected.Value(), actual.Value())
+	}
+	return nil
+}
 
-	return expectedBytes == cgroupMemoryLimitsInBytes
+// compareClaimedStorage compares the requested storage specified in a nodeSet with the actual capacity claimed in the PVC
+func compareClaimedStorage(k8sClient *test.K8sClient, topologyElement esv1.NodeSet, pods []corev1.Pod) error {
```

Comment:
I'm wondering if it would not allow us to remove [`PVCs should eventually be resized`](https://github.com/elastic/cloud-on-k8s/blob/7e71c1bcc56d24bb19e623f3b3ac456b0d4f8724/test/e2e/es/volume_test.go#L198-L213). It is also fine to keep both.

Reply:
Yes, I hesitated on this point, thinking it is better to have too many checks than not enough, but it's redundant so now I feel it doesn't make sense to keep it and we have to trust `compareClaimedStorage`.